architect-team/arcctl
# Serverless GCE + many various bug fixes

## Description

Builds on: https://github.com/architect-team/arcctl/pull/89

## To Test:

```sh
arcctl apply dc DATACENTER_NAME ./examples/datacenters/gcp-serverless.yml
arcctl apply env ENVIRONMENT_NAME -d DATACENTER_NAME
arcctl deploy tyleraldrich/twitter-clone:latest -e ENVIRONMENT_NAME -i app:app
```

When this is finished, the twitter clone service will be available at `http://app.[YOUR DNS ZONE]`.

You should be able to click sign up and redirect to the Kratos login page. Once you enter your signup info, an email will be sent to the smtp4dev service - the DNS name for this is going to be random so... in GCP Cloud DNS you can look at the names generated and find the one for smtp4dev which is deployed via Google Compute Engine, and going to that URL will show you a mailbox with the verification code. After verifying the email, you can login in to the Twitter Clone and send a yeet.


This code is currently compiled, uploaded to a DO bucket, and in use on https://github.com/architect-team/twitter-clone/actions and https://github.com/architect-team/twitter-clone-datacenter/actions to create a datacenter and deploy a component
